raft/server.hh

```diff
@@ -43,8 +45,16 @@ public:
         bool enable_forwarding = true;
 
         // Max size of a single command, add_entry with a bigger command will throw command_is_too_big_error.
-        // The value of zero means no limit.
-        size_t max_command_size = 0;
+        // The following condition must be satisfied:
+        // max_command_memory_usage() <= max_log_size - max_snapshot_trailing_bytes
```

Comment:
It would be nice to explain why.

Reply:
suppose we don't have this rule and consider this scenario:
max_log_size=10
max_command_memory_usage=7
* command of size 4 gets committed/applied, snapshot is not taken, as the limits are not reached
* command of 7 enters `add_entry_on_leader`. It will wait forever, as there is nothing that will trigger the snapshot.

This is weird condition, I agree, and that's why there is a follow-up (more complicated) patch, where `add_entry_on_leader` actively triggers snapshot in this situation.

Reply:
This is why it's necessary to have snapshot_threshold in bytes, and just like with the number of commands, verify that snapshot_threshold + sizeof(new command) < max_log_size with the configruation.

Reply:
this's cleaner, I agree

for group0 we could have
max_command_size = cl->max_record_size() / 2 ~ 32mb
max_log_size = max_command_size;
max_snapshot_trailing_bytes = 1mb
max_snapshot_size = 5mb

max_snapshot_size should include trailing entries, so it's real size would be smaller

Reply:
snapshot_log_size?

Reply:
Is this the threshold to take a snapshot at? Then something like log_snapshoting_threshold_size. Something to indicate what this value is used for.

Reply:
snapshot_threshold_log_size

Reply:
> snapshot_threshold_log_size

ok